Merge `main` @cb8726d

julien-deramond opened the pull request:
- [x] https://github.com/twbs/bootstrap/commit/48807758af6f58e08bb5bc6a27851b68da8f5009
- [ ] https://github.com/twbs/bootstrap/commit/daefd346935d37e7a45bfc1bb2c9108c8ee2faa1
- [ ] https://github.com/twbs/bootstrap/commit/a163ed76d4a0ebd30405f9b351a44497ae7778e1
- [ ] https://github.com/twbs/bootstrap/commit/918a86b42528ec8f7f5d6df71851ab2ec0bc65e7
- [ ] https://github.com/twbs/bootstrap/commit/ef9d8538a06f43f74f49d6829cbf301773381488
- [ ] https://github.com/twbs/bootstrap/commit/85938bb2ed2ec6a97c8dfe9f54c5c450f5c1e86f
- [ ] https://github.com/twbs/bootstrap/commit/3e6265ac55cd8548a83f2470a19ff1dcf28d3b0a
- [ ] https://github.com/twbs/bootstrap/commit/d568e029dac04b50305000df4c1bf33cac7db64a
- [ ] https://github.com/twbs/bootstrap/commit/42da2b95566fe0377543120698e242f4756f648e
- [ ] https://github.com/twbs/bootstrap/commit/2a09f10676cd66aa435fb09c4bc5c19225f10dd9
- [ ] https://github.com/twbs/bootstrap/commit/48a7160cf4e05dc18dde9cf33fa38c12d1bd2e5e
- [ ] https://github.com/twbs/bootstrap/commit/63f75bccd8d206bb59b0e8421258ce03d298b784
- [ ] https://github.com/twbs/bootstrap/commit/5565c64bde8a203560c596c67eebb7a6f2fed412
- [ ] https://github.com/twbs/bootstrap/commit/999bfaa036cb886389050db3645d74dfede0d2e8
- [ ] https://github.com/twbs/bootstrap/commit/d82602063ac48359a0f15f1033837b53f72bef3a
- [ ] https://github.com/twbs/bootstrap/commit/f7f450ec59fd4b7a2e73df5f0eba2dd496edf387
- [ ] https://github.com/twbs/bootstrap/commit/546e34cf85a31a5d553c15afbeef8591d0935efc
- [ ] https://github.com/twbs/bootstrap/commit/ccba6a3589ba33e25ab2919cb844b5bc870e2fef
- [ ] https://github.com/twbs/bootstrap/commit/a247fe9b27ec57152f07aaa5dfc79f633d2b6d10
- [ ] https://github.com/twbs/bootstrap/commit/d97125475b3a5cbbafe3100d5981fd456c92c722
- [ ] https://github.com/twbs/bootstrap/commit/928bdcadc56cef30b4483616c4b2eee7cfa34bd2
- [ ] https://github.com/twbs/bootstrap/commit/d52f6c9de144ac2bc3eba002d3db538183c52465
- [ ] https://github.com/twbs/bootstrap/commit/642d756eeaf107bed2baac931933c9350c33f5c1
- [ ] https://github.com/twbs/bootstrap/commit/407af8ac7f9296627aebc1e4c5d0ee948f8be1f3
- [ ] https://github.com/twbs/bootstrap/commit/ae12d645ef0225831e59248c45689042ebdb83d2
- [ ] https://github.com/twbs/bootstrap/commit/4b17868fb47abf6ed2e994b6c1d0961b34b912a9
- [ ] https://github.com/twbs/bootstrap/commit/353ad45b4b8dd3235a9e26dcc614baaa7fa1a840
- [ ] https://github.com/twbs/bootstrap/commit/cb8726d9e75b10c52c84753da171daacd78aee90

isabellechanclou commented:
Isabelle's comments:
- docs/5.1/components/buttons/Examples > Primary button: Line height is 20px when 18px is indicated in ODS `Important button WEB-BTN-IMP-001` => was already like that, but mentioned here for memory

julien-deramond commented:
> Small issue here : [Close button preview](https://deploy-preview-1108--boosted.netlify.app/docs/5.1/components/close-button/#without-specific-class) on disabled state ([comparison](https://boosted.netlify.app/docs/5.1/components/close-button/#without-specific-class))

Should be fixed with https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1108/commits/0ec87a2f2bf87807253a797639f9a87d47e40cde

julien-deramond commented:
**Note for myself**: rebase and check the quantity selector that has been just merged

julien-deramond commented:
> warning inverse dropdown split buttons are broken

Fixed it with https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1108/commits/72f427bcf2bb9c5d7ec2a8be732155c6cf47d73d. But this modification needs to check non-regression between buttons, dropdowns and split buttons.

julien-deramond commented:
https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/1108/commits/e2ba863f03bda82bc5e06b56ec8a46c135767cfa allows to handle more precisely the `.btn-link`s.

louismaximepiton commented on scss/mixins/_buttons.scss:
```diff
@@ -16,63 +20,45 @@
-  line-height: $line-height;
-  // Manually declare to provide an override to the browser default
-  @include border-radius($border-radius, 0);
+  // Boosted mod: can't use `--#{$variable-prefix}btn-padding-y: #{$padding-y}` here because of different values for `padding-top` and `padding-bottom`
```
I think you handled this case directly l.29 of buttons.scss file, so I think you could remove this part (as Boosted mod at least)

louismaximepiton commented on scss/_buttons.scss:
```diff
@@ -202,6 +204,20 @@
   }
 
+  // Boosted mod
+  &:focus:not(:hover) {
```
I think there's a better way to handle this with `outline-color: var(--#{$variable-prefix}btn-active-border-color);` here instead of `box-shadow`. IMO, it's not an urge.

louismaximepiton reviewed (approved):
LGTM, minor issues can be handled later IMO. 

Minor issues detected : 
- .btn-inverse.btn-no-outline:focus is a bit weird : was already here
- .btn-link outline doesn't move anymore

julien-deramond commented on scss/_button-group.scss:
```diff
@@ -80,8 +80,23 @@
     border-left-color: transparent;
 
-    &.btn-inverse::before {
+    &.btn-inverse {
```
I agree, I add a TODO in the code

julien-deramond commented on scss/mixins/_buttons.scss:
```diff
@@ -16,63 +20,45 @@
   $disabled-border: $gray-500, // Boosted mod
-  $disabled-color: $white // Boosted mod
+  $disabled-color: $white, // Boosted mod
+  $inverted: false // Boosted mod
```
Yes I've done that because I used `$border` and `$hover-color` params. Since it is not the case anymore, I can create a `$btn-focus-shadow-color` param.
